packages/client/src/query/aggregations/GroupByClause.ts

```diff
@@ -25,11 +25,52 @@ export type GroupByClause<
 };
 export type StringGroupByValue = "exact" | { exactWithLimit: number };
 
-export type GroupByRange = [number, number];
+export type GroupByRange<T> = [T, T];
+export type DurationGroupBy = [number, keyof typeof TimeUnitMapping];
```

Comment:
Does this work? Or can I specify any string as a duration time unit? (combined with the satisfies `Record<string, TimeUnit>` below)

Reply:
Additionally it sounds like "YEARS" "QUARTERS" "MONTHS" "WEEKS" all require value to be 1 (see AggregationDurationGroupingV2's jsdoc), which isn't captured in our types.

Reply:
Let me check on the type again, can write a type test for it too just to make sure. And ah thanks for the flag on the years, quarters etc.